libs/optimizer/src/lib/optimizer.ts

```diff
@@ -199,70 +191,91 @@ function toIgcFile(track: ScoringTrack): IGCFile {
+function toOptimizationResult(
+  solution: Solution,
+  originalTrack: ScoringTrack,
+  solverTrack: ScoringTrack,
+): OptimizationResult {
   return {
     score: solution.score ?? 0,
     lengthKm: solution.scoreInfo?.distance ?? 0,
     multiplier: solution.opt.scoring.multiplier,
     circuit: toCircuitType(solution.opt.scoring.code),
-    closingRadius: getClosingRadius(solution),
+    closingRadiusM: getClosingRadiusM(solution),
     solutionIndices: getIndices(solution, originalTrack, solverTrack),
     optimal: solution.optimal || false,
   };
 }
 
-function getClosingRadius(solution: Solution) {
-  // @ts-ignore : closingDistanceFixed is not exposed by library
-  const closingDistanceFixed: number | undefined = solution.opt.scoring?.closingDistanceFixed;
-  // @ts-ignore : closingDistanceRelative is not exposed by library
-  const closingDistanceRelativeRatio: number | undefined = solution.opt.scoring?.closingDistanceRelative;
-  const closingDistanceRelative =
-    solution.scoreInfo?.distance && closingDistanceRelativeRatio
-      ? closingDistanceRelativeRatio * solution.scoreInfo?.distance
-      : undefined;
+// TODO: submit a PR to igc-xc-score
+interface Scoring {
+  closingDistanceFixed?: number;
+  closingDistanceRelative?: number;
+}
+
+function getClosingRadiusM(solution: Solution): number | undefined {
   const closingDistance = solution.scoreInfo?.cp?.d;
+
   if (closingDistance == null) {
     return undefined;
   }
+
+  const closingDistanceFixed = solution.opt.scoring?.closingDistanceFixed;
+
   if (closingDistanceFixed != null && closingDistance < closingDistanceFixed) {
     return closingDistanceFixed;
-  } else if (closingDistanceRelative != null && closingDistance < closingDistanceRelative) {
+  }
+
+  const closingDistanceRelativeRatio = solution.opt.scoring?.closingDistanceRelative;
+  const closingDistanceRelative =
+    solution.scoreInfo?.distance != null && closingDistanceRelativeRatio != null
+      ? closingDistanceRelativeRatio * solution.scoreInfo.distance
+      : undefined;
+
+  if (closingDistanceRelative != null && closingDistance < closingDistanceRelative) {
     return closingDistanceRelative;
   }
+
   return undefined;
 }
 
-const circuitTypeCodes = ['od' , 'tri' , 'fai' , 'oar']
+const circuitTypeCodes = ['od', 'tri', 'fai', 'oar'];
 type CircuitTypeCode = (typeof circuitTypeCodes)[number];
 
-function toCircuitType(code: CircuitTypeCode) {
-  switch (code) {
-    case 'od':
-      return CircuitType.OpenDistance;
-    case 'fai':
-      return CircuitType.FaiTriangle;
-    case 'oar':
-      return CircuitType.OutAndReturn;
-    case 'tri':
-      return CircuitType.FlatTriangle;
+const circuitMapping = {
+  od: CircuitType.OpenDistance,
+  tri: CircuitType.FlatTriangle,
+  fai: CircuitType.FaiTriangle,
+  oar: CircuitType.OutAndReturn,
+};
+
+function toCircuitType(code: string): CircuitType {
+  const type = circuitMapping[code];
```

Comment:
compilation issue
```suggestion
  const type = circuitMapping[code as CircuitTypeCodes];
```
should work